### PR TITLE
Replace the travis scripts with GitHub Actions

### PR DESCRIPTION
**Pull request by ansman**

### CODE REVIEW

**Comment by ansman** [comment]
@gaul This is ready for review now. You can see a successful run here: https://github.com/ansman/s3proxy/runs/4110086097

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          submodules: 'recursive'
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+      - name: Install dependencies
+        run: sudo apt-get install -y libevent-dev python3-pip python3-virtualenv
+      - name: Run Tests Tests
+        run: |
+          mvn package
```
Comment (ansman) [inline .github/workflows/run-tests.yml:18]: I'm not sure why I have to run `mvn package` while the travis script only ran `mvn test` but it seems that `s3proxy` isn't generated when I run `mvn package`

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          ./src/test/resources/run-s3-tests.sh
+      - name: Build Docker Image
+        run: ./publish_img.sh
+        env:
+          DOCKER_USER: ${{ secrets.DOCKER_USER }}
+          DOCKER_PASS: ${{ secrets.DOCKER_PASS }}
```
Comment (ansman) [inline .github/workflows/run-tests.yml:24]: Make sure to add these two secrets before merging

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (gaul) [reply .github/workflows/run-tests.yml:18]: I suspect that this is just some difference between Travis and Docker.  We might want to use `mvn verify` but let's follow up in a subsequent PR.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (gaul) [reply .github/workflows/run-tests.yml:24]: I added secrets.  Let's merge this then try to force a build.

Comment (gaul) [reply .github/workflows/run-tests.yml:24]: Could you take a look at https://github.com/gaul/s3proxy/runs/4211919696?check_suite_focus=true#step:6:18402 ?  I suspect that `REPO` needs to be set.

File: publish_img.sh [hunk 3]
```diff
@@ -1,7 +1,12 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (gaul) [inline publish_img.sh:1]: Might want `set -o unset` here.

**Pull request merged**